moveit_core/trajectory_processing/src/ruckig_traj_smoothing.cpp

```diff
@@ -78,6 +78,33 @@ bool RuckigSmoothing::applySmoothing(robot_trajectory::RobotTrajectory& trajecto
   // This lib does not actually work properly when angles wrap around, so we need to unwind the path first
   trajectory.unwind();
 
+  // Create a copy of the desired trajectory so we can revert it as needed.
+  // Remove repeated waypoints with no change in position. Ruckig does not handle this well and there's really no
+  // need to smooth it. Repeated waypoints cause circular motions.
+  robot_trajectory::RobotTrajectory requested_trajectory(trajectory.getRobotModel(), trajectory.getGroup());
+  for (size_t waypoint_idx = 0; waypoint_idx < num_waypoints - 1; ++waypoint_idx)
+  {
+    bool identical_waypoint = checkForIdenticalWaypoints(
+        *trajectory.getWayPointPtr(waypoint_idx), *trajectory.getWayPointPtr(waypoint_idx + 1), trajectory.getGroup());
+    if (identical_waypoint)
+    {
+      continue;
+    }
+    else
+    {
+      requested_trajectory.addSuffixWayPoint(trajectory.getWayPoint(waypoint_idx),
+                                             trajectory.getWayPointDurationFromPrevious(waypoint_idx));
+    }
```

Comment:
Couple ideas. The `if/else` can be dropped and replaced with 
```
if(!identical_waypoint) {
 requested_trajectory.addSuffixWayPoint(trajectory.getWayPoint(waypoint_idx),
     trajectory.getWayPointDurationFromPrevious(waypoint_idx));
}
```
Logically it is the same of course. Also, the current waypoint, or the pointer to it, can be cached. Net benefit is probably negligible, but it would halve the number of calls to `getWayPointPtr`.

Reply:
Thanks!